File: android/app/src/main/java/com/ccoan/ny/data/models/ApiResponses.kt
package com.ccoan.ny.data.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val error: String? = null,
    val message: String? = null
)

@JsonClass(generateAdapter = true)
data class CreateProjectResponse(
    val id: String,
    val message: String? = null
)

@JsonClass(generateAdapter = true)
data class VapidKeyResponse(
    val configured: Boolean,
    val publicKey: String? = null
)

File: ios/CCOAN/Models/APIResponses.swift
import Foundation

struct APIResponse<T: Decodable>: Decodable {
    let success: Bool
    let data: T?
    let error: String?
    let message: String?
}

struct EmptyResponse: Decodable {}

struct CreateProjectResponse: Decodable {
    let id: String
    let message: String?
}

struct VapidKeyResponse: Decodable {
    let configured: Bool
    let publicKey: String?
}
